wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/expression/values/EngineFunctionEvaluators.scala

```diff
@@ -595,17 +595,19 @@ object EngineFunctionEvaluators {
                                ioFunctionSet: IoFunctionSet,
                                forCommandInstantiationOptions: Option[ForCommandInstantiationOptions])
                               (implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[WomString]] = {
-      def simpleBasename(fileNameAsString: WomString) = fileNameAsString.valueString.split('/').last
+      def simpleBasename(fileNameAsString: WomString): Try[String] = {
+        Try(Await.result(ioFunctionSet.resolvedFileBasename(fileNameAsString.valueString), 60.seconds))
+      }
 
       a.suffixToRemove match {
-        case None => processValidatedSingleValue[WomString, WomString](a.param.evaluateValue(inputs, ioFunctionSet, forCommandInstantiationOptions)) { str =>
-          EvaluatedValue(WomString(simpleBasename(str)), Seq.empty).validNel
+        case None => processValidatedSingleValue[WomString, WomString](a.param.evaluateValue(inputs, ioFunctionSet, forCommandInstantiationOptions)) { filePathString =>
+          simpleBasename(filePathString).map(basename => EvaluatedValue(WomString(basename), Seq.empty)).toErrorOrWithContext(s"interpret '${filePathString.valueString}' as a file path input for basename")
```

Comment:
Without fully analyzing the details, this feels like `validNel' was a valid result before but is not anymore. I don't know what the implications of this are. Can you explain whether or not this is okay?

Reply:
Before:
- Any String passed to `basename` would succeed (no matter the contents) with _some_ value

Now:
- The String must be interpretable as a path (by the currently configured file systems) or else the call will fail